### DISCUSSION TITLE
C# for TCP and UDP client

### DISCUSSION THREAD

**Discussion by Sshisbachza** [nxrighthere/UnrealCLR#228] [Help] [answered]
I'm developing an fps game and I need the proper connections, etc for ue4, but with c++ I'm having a lot of threading and crashing issues. I would like to know if the plugin is ready to do this without any problem, everything is in production. It would be a tcp and udp client, connecting and sending and receiving packets

**Comment by nxrighthere** [comment, marked as answer]
Yes, you can work with TCP and UDP functionality, just like in a regular .NET application.

> everything is in production

The plugin is in the preview stage and not ready for production.
